subprojects/core/src/main/java/org/gradle/api/internal/AbstractPolymorphicDomainObjectContainer.java

```diff
@@ -114,6 +122,44 @@ protected ConfigureDelegate createConfigureDelegate(Closure configureClosure) {
         return new PolymorphicDomainObjectContainerConfigureDelegate<>(configureClosure, this);
     }
 
+    @Override
+    public NamedDomainObjectCollectionSchema getCollectionSchema() {
```

Comment:
:x: this code is now duplicated with `TaskCollection`, this should be fixed before merging this PR

Reply:
A) I would recommend that we don't do it now, because if I move this code to a common parent class (like `DefaultNamedDomainObjectCollection`), then a lot of unrelated siblings could be affected too. I added it instead as a TODO to https://github.com/gradle/gradle/issues/23740

B) Alternatively, we could extract it to some utility class as a temporary solution.

Which one do you prefer?

Reply:
Extracting it to a common place :+1: